packages/components/src/border-control/border-control-dropdown/component.tsx

```diff
@@ -65,68 +66,77 @@ const getColorObject = (
 	return colors.find( ( color ) => color.color === colorValue );
 };
 
-const getToggleAriaLabel = (
+const getToggleDescription = (
 	colorValue: CSSProperties[ 'borderColor' ],
 	colorObject: ColorObject | undefined,
```

Comment:
Does this function _need_ to know so much? Can it not simply accept a colorValue and the _consumer_ can pass the appropriate value. This could reduce the number of confusing conditionals.

Reply:
Oh I see it needs to get the `name` from the object....

Reply:
In which case it could accept 

- `colorValue`
- `colorName`

That way it would not need to know about the structure of a `colorObject`. It can then simply compute appropriate strings based on what you provide it.

Again let's not do this refactoring without tests 🙏